sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py

```diff
@@ -132,17 +132,42 @@ def __init__(
 
         self._populate_attributes(**kwargs)
 
+    # Python 3.5 does not allow for yielding from a coroutine, so instead of the try-finally functional wrapper
+    # trick to restore the timeout, let's use a wrapper class to maintain the override that may be specified.
```

Comment:
I'm a bit confused here although I saw your comment on "Python 3.5 does not allow for yielding from a coroutine".

so we can't get restoring handler timeout in `async def __anext__(self)` of `ServiceBusReceiver`?

Cuz the code I have in my mind is:

```python
class ServiceBusReceiver:
    async def __anext__(self):
        self._check_live()
        original_timeout = None
        try:
            if self.max_wait_time and self.receiver and self.receiver._handler:
                original_timeout = self.receiver._handler._timeout
                self.receiver._handler._timeout = self.max_wait_time * 1000 
            while True:
                try:
                    return await self._do_retryable_operation(self._iter_next)
                except StopAsyncIteration:
                    await self.close()
                    self._message_iter = None
                    raise
        finally:
            if original_timeout:
                self.receiver._handler._timeout = original_timeout
```

This will fall in py 3.5?